### PR TITLE
Check buttons/links have meaningful content

### LINKED ISSUE #34: Buttons and links should have meaningful content
### Describe the new a11y feature or project enhancement

Buttons and links should contain something that indicates what they are or what they are going to go. Having an empty link, or a button that is only an image, or an empty input button without a value can be confusing and doesn't provide enough information for screen reader users.

From [WCAG 2.1 Success Criterion 2.4.4 Link Purpose (In Context)](https://www.w3.org/TR/WCAG21/#link-purpose-in-context): "_The purpose of each link can be determined from the link text alone or from the link text together with its programmatically determined link context [...]_."

### Describe the solution you'd like

Verify that buttons and links follow some basic patterns:

- They are not empty (e.g. using the `:empty` pseudo-class)
- They don't contain just an image tag (e.g. using the `:only-child` pseudo-class)

### Link(s)

- [WCAG 2.1: Success Criterion 2.4.4 Link Purpose (In Context)](https://www.w3.org/TR/WCAG21/#link-purpose-in-context)
- [WebAIM: Links and hypertext](https://webaim.org/techniques/hypertext/link_text#empty_links)

### PR DESCRIPTION
**Pull request by tricinel**
Buttons and links that are not labeled appropriately (either with aria-label or aria-labelledby) and are empty are not of much help to assistive technologies.

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

We're checking for:
- all anchors that have a `href` property and are not labeled and are empty
- all buttons that are not labeled and are empty

These elements break the [Success Criterion 2.4.4 Link Purpose (In Context)](https://www.w3.org/TR/WCAG21/#link-purpose-in-context).

This fixes #34.

## Link(s)

- [https://www.w3.org/TR/WCAG21/#link-purpose-in-context](https://www.w3.org/TR/WCAG21/#link-purpose-in-context)

## Screenshot(s)

Checka11y fix for 34

## Checklist:

- [x] I have thoroughly read the [CONTRIBUTING](../CONTRIBUTING.md) guidelines.
- [x] I understand my pull request will be thoroughly reviewed at high detail.
- [x] I understand the work in my pull request will **only** be available in the next version release of Checka11y.css and not in the current version release.
- [x] I confirm the work in this pull request is valid according to my findings and is not something for anything personal.
- [x] I have updated the [README](../README.md) where and if applicable (still put an `x` if you have considered this but thought there was nothing to add or modify).
- [x] I have added myself to the `contributors` section in `package.json` (still put an `x` if you have considered this but decided not to add yourself).
- [x] I have checked I have **not** committed any **accidental** files.
- [x] I have tested all the main modern browsers (I.e. Chrome, Firefox, Edge, Safari - please leave this unchecked if there were any browsers listed you could not test and list them in the [help](#help) section with details why you couldn't test that browser)
- [x] All new and existing a11y checks still work correctly (compare your local `test/index.html` to the `test/index.html` in the `master` branch).

### CODE REVIEW

File: src/features/_buttons.scss [hunk 1]
```diff
@@ -57,3 +57,11 @@ button {
     @extend %voidError;
   }
 }
+
+/* Buttons have meaningful content */
+button:not( [aria-label] ):not( [aria-labelledby] ):empty {
+  &::after {
+    @extend %contentError;
+    @include contentMessage(error, "Ensure that <button> has meaningful content or is labeled appropriately.");
```
Comment (jackdomleo7) [inline src/features/_buttons.scss:65]: Typo here 😊 - "labelled"

File: src/features/_links.scss [hunk 2]
```diff
@@ -65,3 +65,11 @@ a {
     @extend %voidError;
   }
 }
+
+/* Anchors used as links have meaningful content */
+a[href]:not( [aria-label] ):not( [aria-labelledby] ):empty {
+  &::after {
+    @extend %contentError;
+    @include contentMessage(error, "Ensure that <a> has meaningful content or is labeled appropriately.");
```
Comment (jackdomleo7) [inline src/features/_links.scss:73]: Typo here 😊 - "labelled"

**Review by jackdomleo7 — CHANGES_REQUESTED** [review]
Very nice and clean! Just a few typos then it can be merged. Thank you! 😊

File: src/features/_links.scss (hunk 2 above)
Comment (tricinel) [reply src/features/_links.scss:73]: Ah damn! I'm always unsure which is correct :D Fixing now :) Thanks!

Comment (tricinel) [reply src/features/_links.scss]: That should do it. I spotted the same in the test.

**Review by jackdomleo7 — APPROVED** [review]
Thank you for this!

**Comment by tricinel** [comment]
Welcome! Thanks for taking a look at this so quickly :)

**Pull request merged**